packages/framework/store/src/schema/base.ts

```diff
@@ -206,6 +213,35 @@ export class BlockModel<
 
   childrenUpdated = new Slot();
 
+  childMap = computed(() =>
+    this._children.value.reduce((map, id, index) => {
+      map.set(id, index);
+      return map;
+    }, new Map<string, number>())
+  );
+
+  isEmpty = computed(() => {
+    return this._children.value.length === 0;
+  });
+
+  constructor() {
+    super();
+    this._onCreated = this.created.once(() => {
+      this._children.value = this.yBlock.get('sys:children').toArray();
+      this.yBlock.get('sys:children').observe(event => {
+        this._children.value = event.target.toArray();
+      });
+    });
+    this._onDeleted = this.deleted.once(() => {
+      this._onCreated.dispose();
+    });
```

Comment:
I think these two may be initialized in class properties as well instead of being syntactically in the constructor.

Reply:
Is there any special reason for this?

Reply:
I think .value actually subscribes to the signal in one doc that I read. Let me double check.